### PR TITLE
Node cron

### LINKED ISSUE #2: Create a node-cron service
To remind patients to take the medicines on time.

### PR DESCRIPTION
**Pull request by sbk2k1**
issue: #2  
I did not know about the time/frequency format of the routine models. Hence I created a node-cron-job to run every minute to check for medicines within the start and end time period and return only relevant details back for notifying users.

### CODE REVIEW

File: services/cron.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+const cron = require('node-cron');
+const Routine = require('../models/routine.model');
+
+// run a cron job every minute
+export const checkEveryMinute= cron.schedule('* * * * *', async () => {
+    console.log('running a task every minute');
+
+    // Since I did not know about the time format I return medicines between the start and end time
```
Comment (Codehackerone) [inline services/cron.ts]: please dont write 'I'

File: services/cron.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+const cron = require('node-cron');
+const Routine = require('../models/routine.model');
+
+// run a cron job every minute
+export const checkEveryMinute= cron.schedule('* * * * *', async () => {
```
Comment (Codehackerone) [inline services/cron.ts]: Are you sure this runs crobjob every one minute?

File: services/cron.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    // Since I did not know about the time format I return medicines between the start and end time
+    const drugsToBeNotified = await Routine.aggregate([
+        {
+            "$match": {
+                //matching the current time with the start and end time of the medicine
+                "medicineStartTime": {
+                    "$lte": new Date().getTime()
+                },
+                "medicineEndTime": {
+                    "$gte": new Date().getTime()
```
Comment (Codehackerone) [inline services/cron.ts]: These are not the correct fields.. add the time in the model. Create a new issue if you have to.

**Review by Codehackerone — CHANGES_REQUESTED** [review]
Delete package-lock.

File: services/cron.ts (hunk 2 above)
Comment (sbk2k1) [reply services/cron.ts]: yes

**Review by Codehackerone — COMMENTED** [review]
Seems alright! Did you test this functionality? If yes, can you share some ss?

**Comment by Codehackerone** [comment]
Okay no need... seems to work alright.

**Pull request merged**